src/UsgsAstroSarSensorModel.cpp

```diff
@@ -513,47 +635,64 @@ csm::ImageCoord UsgsAstroSarSensorModel::getImageStart() const
 
 csm::ImageVector UsgsAstroSarSensorModel::getImageSize() const
 {
-  return csm::ImageVector(0.0, 0.0);
+  return csm::ImageVector(m_nLines, m_nSamples);
 }
 
 pair<csm::ImageCoord, csm::ImageCoord> UsgsAstroSarSensorModel::getValidImageRange() const
 {
-  return make_pair(csm::ImageCoord(0.0, 0.0), csm::ImageCoord(0.0, 0.0));
+  csm::ImageCoord start = getImageStart();
+  csm::ImageVector size = getImageSize();
+  return make_pair(start, csm::ImageCoord(start.line + size.line, start.samp + start.samp));
 }
 
 pair<double, double> UsgsAstroSarSensorModel::getValidHeightRange() const
 {
-  return make_pair(0.0, 0.0);
+  return make_pair(m_minElevation, m_maxElevation);
 }
 
 csm::EcefVector UsgsAstroSarSensorModel::getIlluminationDirection(const csm::EcefCoord& groundPt) const
 {
-  return csm::EcefVector(0.0, 0.0, 0.0);
+  csm::EcefVector sunPosition = getSunPosition(getImageTime(groundToImage(groundPt)));
+  csm::EcefVector illuminationDirection = csm::EcefVector(groundPt.x - sunPosition.x,
```

Comment:
Might as well use the utility functions you just added to do vector operations? 

like `illuminationDirection = groundPt - sunPosition`?  But then I guess you'd need to make a vector for groundPt. Not sure whether it's worth it or not.

Reply:
And norm(illuminationDirection) below.

Reply:
Yeah all of this was written before I added those. I need to go back over it and use those!